main_test.go

```diff
@@ -549,6 +594,80 @@ func TestProxy1(t *testing.T) {
 
 var errTest = errors.New("test")
 
+func TestProxy2(t *testing.T) {
+	var cancelfunc context.CancelFunc
+	Convey("a setup signalfx gateway", t, func() {
+		sendTo := dptest.NewBasicSink()
+		var ctx context.Context
+		ctx, cancelfunc = context.WithCancel(context.Background())
+		var p *gateway
+		var mainDoneChan chan error
+		var filename string
+		var sl *signalfx.ListenerServer
+		checkError := false
+
+		setUp := func(max int, min int, check int, internalMetricsAddress string) {
+			logBuf := &ConcurrentByteBuffer{&bytes.Buffer{}, sync.Mutex{}}
+			logger := log.NewHierarchy(log.NewLogfmtLogger(io.MultiWriter(logBuf, os.Stderr), log.Panic))
+			fileObj, err := ioutil.TempFile("", "TestProxy")
+			So(err, ShouldBeNil)
+			etcdDataDir, err := ioutil.TempDir("", "TestProxy2")
+			So(err, ShouldBeNil)
+			So(os.RemoveAll(etcdDataDir), ShouldBeNil)
+			filename = fileObj.Name()
+			So(os.Remove(filename), ShouldBeNil)
+			sconf := &signalfx.ListenerConfig{}
+			sl, err = signalfx.NewListener(sendTo, sconf)
+			So(err, ShouldBeNil)
+			openPort := nettest.TCPPort(sl)
+			proxyConf := strings.Replace(config2, "<<PORT>>", strconv.FormatInt(int64(openPort), 10), -1)
+			proxyConf = strings.Replace(proxyConf, "<<MAX>>", strconv.FormatInt(int64(max), 10), -1)
+			proxyConf = strings.Replace(proxyConf, "<<MIN>>", strconv.FormatInt(int64(min), 10), -1)
+			proxyConf = strings.Replace(proxyConf, "<<CHECK>>", strconv.FormatInt(int64(check), 10), -1)
+			proxyConf = strings.Replace(proxyConf, "<<INTERNALMETRICS>>", internalMetricsAddress, -1)
+			So(ioutil.WriteFile(filename, []byte(proxyConf), os.FileMode(0666)), ShouldBeNil)
+			fmt.Println("Launching server...")
+			p = newGateway()
+			p.logger = logger
+			loadedConfig, _ := loadConfig(filename, logger)
+			p.configure(loadedConfig)
+			mainDoneChan = make(chan error)
+			go func() {
+				mainDoneChan <- p.start(ctx)
+				close(mainDoneChan)
+			}()
+			<-p.setupDoneSignal
+		}
```

Comment:
Getting 
http: panic serving 127.0.0.1:49484: runtime error: invalid memory address or nil pointer dereference 
after running this test